content/blog/gpg-signed-releases/index.md

```diff
@@ -0,0 +1,121 @@
+out the information for Linux/x64 and download the required artifacts - the
+product download and the GPG signature link - using the `curl` command
+(Examples for linux/x64, change accordingly for other platforms)
+
+```
+jq '.[0].binaries[] | select(.architecture=="x64") | select (.os=="linux") | select (.image_type=="jdk") .package.link' adopt.json | xargs curl -LO
+jq '.[0].binaries[] | select(.architecture=="x64") | select (.os=="linux") | select (.image_type=="jdk") .package.signature_link' adopt.json | xargs curl -LO
+```
+
+Note that as mentioned in the previous section, we have previously also
+provided the sha256sums which can be obtained from the JSON file with
+
+```
+jq '.[0].binaries[] | select(.architecture=="x64") | select (.os=="linux") | select (.image_type=="jdk") .package.checksum' adopt.json
+```
+
+If you have not used them before, The SHA checksums can be verified against
+the output from running the `sha256sum` command against the binary artifact.
+The SHA checksum verifies that the downloads have occurred without errors,
+but the GPG checksums additionally verify the source of the binaries is from
+the Adoptium project
+
+## How do I verify the signatures once I have them?
+
+You will need to have the `gpg` tool installed in order to verify the
+signatures.  You can then run the following command to check the signature by supplying the signature file and the corresponding file which the signature is for e.g.:
+
+`gpg --verify OpenJDK17U-jdk_x64_linux_hotspot_17.0.4_8.tar.gz.sig OpenJDK17U-jdk_x64_linux_hotspot_17.0.4_8.tar.gz`
+
+If you do not currently have the public signing key you will get a message such as this:
+
+```
+gpg: directory '/home/sxa/.gnupg' created
+gpg: keybox '/home/sxa/.gnupg/pubring.kbx' created
+gpg: Signature made Mon Jul  4 18:20:31 2022 UTC
+gpg:                using RSA key 3B04D753C9050D9A5D343F39843C48A565F8F04B
+gpg: Can't check signature: No public key
+```
+
+To resolve this message you need to acquire the public key that was used to
+sign the binaries.  You can download it from a trusted GPG server, for
+example to use the Ubuntu key servers run this command:
+
+`gpg --keyserver keyserver.ubuntu.com --recv-keys 3B04D753C9050D9A5D343F39843C48A565F8F04B`
+
+If you then run the verify command you will get a message indicating that the newly imported key has not been trusted:
+
+```
+gpg: Good signature from "Adoptium GPG Key (DEB/RPM Signing Key)
+<temurin-dev@eclipse.org>" [unknown] gpg: WARNING: This key is not certified
+with a trusted signature!  gpg: There is no indication that the signature
+belongs to the owner.
+```
+While the "Good signature" message gives you some
+confidence that the download is valid, to fully trust the certificate and
+remove the final warning you can run the following then follow the prompts
+to grant ultimate trust to it:
+
+```
+gpg --edit-key 3B04D753C9050D9A5D343F39843C48A565F8F04B trust
```

Comment:
I personally wouldn't just trust a key that I got in the keystore this way - even if the unhackable Adoptium blog post ;-) has the fingerprint spelled out this way. But I get it that you probably don't want to get into ways to trust a key. People reading this post who don't understand web of trust probably wouldn't go to the effort of checking a key the same way as me anyway!

Reply:
I completely agree :-) But it's likely the most common way people would do it and is the easiest to give to people as a process they can easily follow. I was leaving those sorts of details to other articles such as the Eclipse one referenced at the end.